plugins/module_utils/network/sonic/facts/lst/lst.py

```diff
@@ -0,0 +1,156 @@
+        objs = data
+        facts = {}
+        if objs:
+            params = utils.validate_config(self.argument_spec, {'config': objs})
+            facts['lst'] = remove_empties(params['config'])
+        ansible_facts['ansible_network_resources'].update(facts)
+        return ansible_facts
+
+    def get_config(self, module):
+        cfg = None
+        get_path = 'data/openconfig-lst-ext:lst'
+        request = {'path': get_path, 'method': 'get'}
+
+        try:
+            response = edit_config(module, to_request(module, request))
+            if 'openconfig-lst-ext:lst' in response[0][1]:
+                cfg = response[0][1].get('openconfig-lst-ext:lst')
+        except ConnectionError as exc:
+            module.fail_json(msg=str(exc), code=exc.code)
+
+        return cfg
+
+    def update_lst(self, cfg):
+        config_dict = {}
+        enum_dict = {'openconfig-lst-ext:GROUP_L3': 'l3', 'openconfig-lst-ext:ONLINE_PERCENTAGE': 'percentage'}
+
+        if cfg:
+            if 'lst-groups' in cfg and 'lst-group' in cfg['lst-groups']:
+                lst_group = cfg['lst-groups']['lst-group']
+                lst_groups_list = []
+                for group in lst_group:
+                    group_dict = {}
+                    name = group.get('name')
+                    config = group.get('config')
+
+                    if name:
+                        group_dict['name'] = name
+                    if config:
+                        # For all_evpn_es_downstream and all_mclags_downstream attributes, current SONiC behavior treats none value as false
+                        all_evpn_es_downstream = config.get('all-evpn-es-downstream', False)
+                        all_mclags_downstream = config.get('all-mclags-downstream', False)
+                        group_description = config.get('description')
+                        group_type = config.get('type')
+                        threshold_down = config.get('threshold-down')
+                        threshold_type = config.get('threshold-type')
+                        threshold_up = config.get('threshold-up')
+                        timeout = config.get('timeout')
+
+                        group_dict['all_evpn_es_downstream'] = all_evpn_es_downstream
+                        group_dict['all_mclags_downstream'] = all_mclags_downstream
+                        if group_description:
+                            group_dict['group_description'] = group_description
+                        if group_type:
+                            group_dict['group_type'] = enum_dict[group_type]
+                        if threshold_down is not None:
+                            group_dict['threshold_down'] = int(threshold_down)
+                        if threshold_type:
+                            group_dict['threshold_type'] = enum_dict[threshold_type]
+                        if threshold_up is not None:
+                            group_dict['threshold_up'] = int(threshold_up)
```

Comment:
This will be okay (only) if the argspec and module definition of the type for "threshold_up" is changed to "int".

(I have posted comments requesting that change.)